### PR TITLE
Migrate to updated JupyterLab extension template

### PR DESCRIPTION
**Pull request by akisaini**
Migrate to the new [extension template](https://github.com/jupyterlab/extension-template) as suggested by @fcollonval in #217

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,56 +2,97 @@ name: Build
 
 on:
   push:
-    branches: master
+    branches: main
```
Comment (ktaletsk) [inline .github/workflows/build.yml:5]: Change this to `master` since we don't have `main` branch (yet)

File: .github/workflows/check-release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Check Release
+on:
+  push:
+    branches: ["main"]
```
Comment (ktaletsk) [inline .github/workflows/check-release.yml:4]: Change this to `master` since we don't have `main` branch (yet)

File: CHANGELOG.md
Comment (ktaletsk) [inline CHANGELOG.md:1]: Please, move the changelog items from `docs/changelog.md` into here

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -2,56 +2,97 @@ name: Build
...
+      uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+    - name: Install dependencies
+      run: python -m pip install -U "jupyterlab>=4.0.0,<5"
+
+    - name: Lint the extension
+      run: |
+        set -eux
+        jlpm
+        jlpm run lint:check
```
Comment (ktaletsk) [inline .github/workflows/build.yml:31]: This step fails in GitHub Actions with `Usage Error: Couldn't find a script named "lint:check".`

File: schema/plugin.json [hunk 4]
```diff
@@ -1,8 +1,10 @@
 {
+  "jupyter.lab.shortcuts": [],
   "jupyter.lab.setting-icon-class": "jp-FileIcon",
   "jupyter.lab.setting-icon-label": "LaTeX",
-  "title": "LaTeX",
-  "description": "LaTeX settings.",
+  "title": "jupyterlab_latex",
+  "description": "jupyterlab_latex settings.",
```
Comment (ktaletsk) [inline schema/plugin.json:6]: Let's change it back to "LaTeX" and "LaTeX settings"

File: tslint.json
Comment (ktaletsk) [inline tslint.json:1]: I wonder if this file is still needed for linter to work (see GitHub Actions build failure)

File: package.json [hunk 5]
```diff
@@ -110,17 +104,97 @@
...
+  "publishConfig": {
+    "access": "public"
+  },
   "jupyterlab": {
-    "extension": "lib/index.js",
+    "discovery": {
+      "server": {
+        "managers": ["pip"],
+        "base": {
+          "name": "@jupyterlab/latex"
```
Comment (ktaletsk) [inline package.json:117]: This seems incorrect to me
The template has it as `{{python_name}}`
```
"discovery": {
            "server": {
                "managers": [
                    "pip"
                ],
                "base": {
                    "name": "{{ python_name }}"
                }
            }
        },
```
Is `@jupyterlab/latex` name of the Python package?

File: .copier-answers.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Changes here will be overwritten by Copier; NEVER EDIT MANUALLY
+_commit: v4.3.5
+_src_path: https://github.com/jupyterlab/extension-template
+author_email: jupyter@googlegroups.com
+author_name: Jupyter Development Team
+has_binder: true
+has_settings: true
+kind: server
+labextension_name: jupyterlab_latex
```
Comment (ktaletsk) [inline .copier-answers.yml:9]: This does not look right. Shouldn't it be `@jupyterlab/latex`?

File: .copier-answers.yml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+project_short_description: JupyterLab extension for running LaTeX
+python_name: jupyterlab_latex
```
Comment (ktaletsk) [inline .copier-answers.yml:11]: Python package has a dash in the middle `jupyterlab-latex`

**Review by ktaletsk — CHANGES_REQUESTED** [review]
Given the incorrect names used in Copier, could you start from the beginning using the template?

File: package.json (hunk 5 above)
Comment (akisaini) [reply package.json:117]: You're right. It's been fixed.

File: .copier-answers.yml (hunk 6 above)
Comment (akisaini) [reply .copier-answers.yml:9]: This file is generated by copier and mentions "# Changes here will be overwritten by Copier; NEVER EDIT MANUALLY" 
I am not entirely sure why is mentions the "labextension_name" as the node package name. Will check.

Comment (akisaini) [reply .copier-answers.yml]: Fixed. 
labextension_name typically follows the format `@<organization>/<package-name>` for scoped packages or just `<package-name>` for non-scoped packages.

File: tslint.json
Comment (akisaini) [reply tslint.json:1]: I fixed package.json and added changes to the scripts section. It should hopefully pass through the `lint:check` step.

**Review by ktaletsk — APPROVED** [review]

**Pull request merged**